src/inet/tests/TestInetEndPoint.cpp

```diff
@@ -506,16 +506,18 @@ static void TestInetEndPointLimit(nlTestSuite * inSuite, void * inContext)
         err = gInet.NewTCPEndPoint(&testTCPEP);
     NL_TEST_ASSERT(inSuite, err == INET_ERROR_NO_ENDPOINTS);
 
+    System::Timer * lTimer = nullptr;
+
     // Verify same aComplete and aAppState args do not exhaust timer pool
     for (int i = 0; i < CHIP_SYSTEM_CONFIG_NUM_TIMERS + 1; i++)
     {
-        err = gSystemLayer.StartTimer(10, HandleTimer, nullptr);
-        NL_TEST_ASSERT(inSuite, err == CHIP_SYSTEM_NO_ERROR);
+        lTimer = gSystemLayer.StartTimer(10, HandleTimer, nullptr);
+        NL_TEST_ASSERT(inSuite, lTimer != nullptr);
     }
 
     for (int i = 0; i < CHIP_SYSTEM_CONFIG_NUM_TIMERS + 1; i++)
-        err = gSystemLayer.StartTimer(10, HandleTimer, &numTimersTest[i]);
-    NL_TEST_ASSERT(inSuite, err == CHIP_SYSTEM_ERROR_NO_MEMORY);
+        lTimer = gSystemLayer.StartTimer(10, HandleTimer, &numTimersTest[i]);
```

Comment:
This tries to create NUM_TIMERS again and only validates the last attempt of creation.
I would drop this for loop and only try 'one last time'. We could put the assert in the loop as well, but that seems wasteful.

Why are we passing in the &numTimersTest[i] here as an argument if we expect failure?

Reply:
The for loop is used to drain the timer pool and we need provide different appState(&numTimersTest[i]) to start the different timers.
I added following code in the for loop and we only need to check the final result.
        if (lTimer == nullptr)            break;